src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        T item;
+        Node<T> prev;
+        Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        Node<T> currentNode = getNodeByIndex(index);
+        if (size == 0) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, currentNode);
+            currentNode.prev = newNode;
+            head = newNode;
+            size++;
+        } else if (size == index) {
+            add(value);
+        } else {
+            Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+            currentNode.prev.next = newNode;
+            currentNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkingIndex(index);
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkingIndex(index);
+        Node<T> currentNode = getNodeByIndex(index);
+        T currentItem = currentNode.item;
+        currentNode.item = value;
+        return currentItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkingIndex(index);
```

Comment:
we can simplify this code too :
```
        checkingIndex(index);
        unlinkNode(getNodeByIndex(index));
```
 method unlinkNode() can be used in the method remove by index and remove by value:
```
private T unlinkNode(Node<T> currentNode){
        Node<T> prev = currentNode.prev;
        Node<T> next = currentNode.next;
        if (prev == null) {
            head = next;
        } else {
            prev.next = next;
        }
        if (next == null) {
            tail = prev;
        } else {
            next.prev = prev;
        }
        size--;
        return currentNode.item;
}
```